File: php/src/Services/MerchantWithdrawService.php
<?php

namespace CCPayment\Services;

use CCPayment\Client;

class MerchantWithdrawService
{
    private Client $client;
    public function __construct(Client $client) { $this->client = $client; }
    public function getCwalletUserId(string $cwalletUserId): array { return $this->client->post('/getCwalletUserId', ['cwalletUserId' => $cwalletUserId]); }
    public function getWithdrawFee(int $coinId, string $chain): array { return $this->client->post('/getWithdrawFee', ['coinId' => $coinId, 'chain' => $chain]); }
    public function applyAppWithdrawToNetwork(array $data): array { return $this->client->post('/applyAppWithdrawToNetwork', $data); }
    public function applyAppWithdrawToCwallet(array $data): array { return $this->client->post('/applyAppWithdrawToCwallet', $data); }
    public function getAppWithdrawRecord(array $data): array { return $this->client->post('/getAppWithdrawRecord', $data); }
    public function getAppWithdrawRecordList(array $data = []): array { return $this->client->post('/getAppWithdrawRecordList', $data); }
    public function getAutoWithdrawRecordList(array $data = []): array { return $this->client->post('/getAutoWithdrawRecordList', $data); }
    public function getRiskRefundRecords(array $data = []): array { return $this->client->post('/getRiskyRefundRecordList', $data); }
}

File: ruby/lib/ccpayment/services/merchant_withdraw_service.rb
module CCPayment
  module Services
    class MerchantWithdrawService
      def initialize(client); @client = client; end
      def get_cwallet_user_id(cwallet_user_id); @client.post('/getCwalletUserId', { cwalletUserId: cwallet_user_id }); end
      def get_withdraw_fee(coin_id, chain); @client.post('/getWithdrawFee', { coinId: coin_id, chain: chain }); end
      def apply_app_withdraw_to_network(data); @client.post('/applyAppWithdrawToNetwork', data); end
      def apply_app_withdraw_to_cwallet(data); @client.post('/applyAppWithdrawToCwallet', data); end
      def get_app_withdraw_record(data); @client.post('/getAppWithdrawRecord', data); end
      def get_app_withdraw_record_list(data = {}); @client.post('/getAppWithdrawRecordList', data); end
      def get_auto_withdraw_record_list(data = {}); @client.post('/getAutoWithdrawRecordList', data); end
      def get_risk_refund_records(data = {}); @client.post('/getRiskyRefundRecordList', data); end
    end
  end
end
